### DISCUSSION TITLE
What is `TlsConfiguration::Simple.ca`?

### DISCUSSION THREAD

**Discussion by J-Kappes** [bytebeamio/rumqtt#903] [Q&A]
[The docs](https://docs.rs/rumqttc/latest/rumqttc/enum.TlsConfiguration.html) say it's a "connection method". Its type is `Vec<u8>`.

**Comment by swanandx** [comment]
Good catch! It should be "ca certificate", if possible can you please open PR to fix it?

Thanks :)

**Reply by J-Kappes** [reply]
OK, I did. But I'm genuinely asking what I'm supposed to put there. Is it supposed to be a byte string of a path to a cert file?

I'm trying to do an encrypted publish to a server with a self-signed certificate which I don't want to verify. I tried passing an empty vector here, but that didn't work. (I already implemented a custom "verifier").

**Reply by swanandx** [reply]
> OK, I did. But I'm genuinely asking what I'm supposed to put there. Is it supposed to be a byte string of a path to a cert file?
> 
It's supposed to be whole CA certificate as bytes. Basically you need to read your certificate as Vec<u8> and pass there. ( you can use something like `include_bytes!("path/to/ca_cert")` or just read_file, check tls2 example )

> I'm trying to do an encrypted publish to a server with a self-signed certificate which I don't want to verify. I tried passing an empty vector here, but that didn't work. (I already implemented a custom "verifier").
>
So I'm assuming you mean you don't want client authentication right? It is optional so you can set it to None. Just specifying ca cert should be enough :)

**Reply by J-Kappes** [reply]
No, it's really the server who has a self-signed certificate and whom I don't want to verify.

I've now noticed that native-tls allows for accepting invalid certs, but the `.into()` call (TlsConnector to TlsConfiguration) used in the example doesn't seem to exist.
